web/app/js/components/Sidebar.jsx

```diff
@@ -202,7 +192,57 @@ class Sidebar extends React.Component {
               <Menu.Item><PrefixedLink to="/pods">Pods</PrefixedLink></Menu.Item>
               <Menu.Item><PrefixedLink to="/replicationcontrollers">Replication Controllers</PrefixedLink></Menu.Item>
             </Menu.SubMenu>
+            <Menu.Divider />
+          </Menu>
+
+          <Menu
+            className="sidebar-menu"
+            theme="dark"
+            mode="inline"
+            selectedKeys={[normalizedPath]}>
+            <Menu.Item className="sidebar-menu-item" key="/namespace-selector">
+              <Form layout="inline">
+                <Form.Item>
+                  <Select
+                    defaultValue="All Namespaces"
+                    dropdownMatchSelectWidth={true}
+                    onChange={this.handleNamespaceSelector}>
+                    {
+                      _.map(namespaces, ns => {
+                        return (
+                          <Select.Option key={ns.nsValue} value={ns.nsValue}>{ns.nsName}</Select.Option>
```

Comment:
`ns.nsValue` is a little repetitive, maybe `ns.labelValue` and `ns.labelName`? or simply `ns.name` and `ns.value`?
`_.map(namespaces, label => ... label.name .... label.value` might also be a clearer option